recipes/g3log/all/conanfile.py

```diff
@@ -0,0 +1,104 @@
+    _source_subfolder = "source_subfolder"
+
+    def _has_support_for_cpp14(self):
+        supported_compilers = [("apple-clang", 5.1), ("clang", 3.4), ("gcc", 6.1), ("Visual Studio", 15.0)]
+        compiler, version = self.settings.compiler, Version(self.settings.compiler.version)
+        return any(compiler == sc[0] and version >= sc[1] for sc in supported_compilers)
+
+    def configure(self):
+        if self.settings.compiler.get_safe("cppstd"):
+            tools.check_min_cppstd(self, "14")
+        if not self._has_support_for_cpp14():
+            raise ConanInvalidConfiguration("g3log requires C++14 or higher support standard."
+                                            " {} {} is not supported."
+                                            .format(self.settings.compiler,
+                                                    self.settings.compiler.version))
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        dir_postfix = self.conan_data["sources"][self.version]["url"].split("/")[-1][:-7]
+        os.rename("g3log-{}".format(dir_postfix), self._source_subfolder)
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+        if self.settings.compiler != "Visual Studio":
+            del self.options.enable_vectored_exception_handling
+            del self.options.debug_break_at_fatal_signal
+
+    def _configure_cmake(self):
+        cmake = CMake(self)
+        cmake.definitions["G3_SHARED_LIB"] = self.options.shared
+        cmake.definitions["USE_DYNAMIC_LOGGING_LEVELS"] = self.options.use_dynamic_logging_levels
+        cmake.definitions["CHANGE_G3LOG_DEBUG_TO_DBUG"] = self.options.change_debug_to_dbug
+        cmake.definitions["USE_G3_DYNAMIC_MAX_MESSAGE_SIZE"] = self.options.use_dynamic_max_message_size
+        cmake.definitions["G3_LOG_FULL_FILENAME"] = self.options.log_full_filename
+        cmake.definitions["ENABLE_FATAL_SIGNALHANDLING"] = self.options.enable_fatal_signal_handling
+        if self.settings.compiler == "Visual Studio":
+            cmake.definitions["ENABLE_VECTORED_EXCEPTIONHANDLING"] = self.options.enable_vectored_exception_handling
+            cmake.definitions["DEBUG_BREAK_AT_FATAL_SIGNAL"] = self.options.debug_break_at_fatal_signal
+        cmake.definitions["ADD_FATAL_EXAMPLE"] = "OFF"
+        cmake.definitions["ADD_G3LOG_PERFORMANCE"] = "OFF"
+        cmake.definitions["ADD_G3LOG_UNIT_TEST"] = "OFF"
+        cmake.configure()
+        return cmake
+
+    def build(self):
+        if "patches" in self.conan_data and self.version in self.conan_data["patches"]:
+            for patch in self.conan_data["patches"][self.version]:
+                tools.patch(**patch)
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def package(self):
+        self.copy("LICENSE", dst="licenses", src=self._source_subfolder)
+        if self.settings.os == "Windows":
```

Comment:
Did you try `cmake.install()` ?

Reply:
Yes, that was my first approach. With version 1.3.2 I struggled quite a bit with link errors, wrong CMAKE_INSTALL_LIBDIR, tried to patch, still errors and then I went for the current fallback solution. The CMake part of g3log as improved quite a bit to version 1.3.2-86, and with that version `cmake.install()` works. But I get some hook errors: `[CMAKE-MODULES-CONFIG-FILES (KB-H016)]` and `[CMAKE FILE NOT IN BUILD FOLDERS (KB-H019)]`.

Reply:
Understood. These errors are regular, the regular approach is running cmake.install() and removing the folders which are not acceptable in CCI: https://github.com/conan-io/conan-center-index/blob/265958ce30706b359000b42dab360687175560a7/recipes/ogg/all/conanfile.py#L47

Reply:
I gave it another try this evening, and this time I was able to fix it. I had to override CMAKE_INSTALL_LIBDIR with CACHE FORCE and change type from PATH to STRING. It now works locally for me with cmake.install() for both version 1.3.2 and 1.3.2-86. I have pushed the changes and hopefully it works for the other platforms as well.

Reply:
I finally got it to work with `cmake.install()`. Fixed in e1e23cb.